src/meta/api/src/schema_api_impl.rs

```diff
@@ -1071,6 +1082,202 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
+                    succ = display(succ),
+                    "create_virtual_column"
+                );
+
+                if succ {
+                    break;
+                }
+            }
+        }
+
+        Ok(CreateVirtualColumnReply {})
+    }
+
+    async fn update_virtual_column(
+        &self,
+        req: UpdateVirtualColumnReq,
+    ) -> Result<UpdateVirtualColumnReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let ctx = &func_name!();
+        let mut trials = txn_trials(None, ctx);
+        loop {
+            trials.next().unwrap()?;
+
+            let (seq, old_virtual_column_meta) =
+                get_virtual_column_by_id_or_err(self, &req.name_ident, ctx).await?;
+
+            let virtual_column_meta = VirtualColumnMeta {
+                table_id: req.name_ident.table_id,
+                virtual_columns: req.virtual_columns.clone(),
+                created_on: old_virtual_column_meta.created_on,
+                updated_on: Some(Utc::now()),
+            };
+
+            // Update virtual column by inserting this record:
+            // (tenant, table_id) -> virtual_column_meta
+            {
+                let condition = vec![txn_cond_seq(&req.name_ident, Eq, seq)];
+                let if_then = vec![txn_op_put(
+                    &req.name_ident,
+                    serialize_struct(&virtual_column_meta)?,
+                )];
+
+                let txn_req = TxnRequest {
+                    condition,
+                    if_then,
+                    else_then: vec![],
+                };
+
+                let (succ, _responses) = send_txn(self, txn_req).await?;
+
+                debug!(
+                    req.name_ident = debug(&virtual_column_meta),
+                    succ = display(succ),
+                    "update_virtual_column"
+                );
+
+                if succ {
+                    break;
+                }
+            }
+        }
+
+        Ok(UpdateVirtualColumnReply {})
+    }
+
+    async fn drop_virtual_column(
+        &self,
+        req: DropVirtualColumnReq,
+    ) -> Result<DropVirtualColumnReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let ctx = &func_name!();
+        let mut trials = txn_trials(None, ctx);
+        loop {
+            trials.next().unwrap()?;
+
+            let (seq, _) = get_virtual_column_by_id_or_err(self, &req.name_ident, ctx).await?;
+
+            // Drop virtual column by deleting this record:
+            // (tenant, table_id) -> virtual_column_meta
+            {
+                let condition = vec![txn_cond_seq(&req.name_ident, Eq, seq)];
+                let if_then = vec![txn_op_del(&req.name_ident)];
+                let txn_req = TxnRequest {
+                    condition,
+                    if_then,
+                    else_then: vec![],
+                };
```

Comment:
If it just the record with a seq that is just fetched, then it does not have to assert the seq in the `condition`. 
Asserting the `seq` is only useful when the seq is passed in by callers.

Reply:
I didn't understand it clearly, let me delete them